amd_gpu/gpu.c

```diff
@@ -163,10 +163,12 @@ const char* err_to_str(cl_int ret)
 		return "CL_INVALID_LINKER_OPTIONS";
 	case CL_INVALID_DEVICE_PARTITION_COUNT:
 		return "CL_INVALID_DEVICE_PARTITION_COUNT";
+#ifndef __APPLE__
```

Comment:
Unfortunately I won't be able to spend too much time on this in the near future so the PR may end up pulling master a few times anyway.